main/src/main/scala/sbt/Defaults.scala

```diff
@@ -363,7 +363,14 @@ object Defaults extends BuildCommon {
   lazy val outputConfigPaths = Seq(
     classDirectory := crossTarget.value / (prefix(configuration.value.name) + "classes"),
     target in doc := crossTarget.value / (prefix(configuration.value.name) + "api")
+  ) ++ inTask(compile)(
```

Comment:
To reviewers: this is scoped in the default configs `Compile`, `Test` and `Runtime`.

Reply:
I should also add `cleanKeepFiles` here and fix dependency on it in `cleanFilesTask`, because otherwise user cannot override it in this particular scope.